shared/rakefiles/secrets.rb

```diff
@@ -67,8 +69,19 @@ def self.collect_secrets()
       end
     end
 
-    encryption_keys.uniq!
-    ENV["TF_VAR_encryption_keys"] = %Q|[ #{encryption_keys.join(", ")} ]|
+    encryption_keys = {}
+    secrets_config = YAML.load(File.read("./modules/gcp-secret-mgmt/config.yaml"))
+    secrets_config["encryption_keys"].each do |encryption_key|
+      encryption_keys[encryption_key] = %Q|"#{encryption_key}"|
+    end
+
+    unless collected_secrets.keys == encryption_keys.keys
+      puts "ERROR: Secret keys: #{(collected_secrets.keys - encryption_keys.keys).join(", ")} not present in"
+      puts "ERROR: /modules/gcp-secret-mgmt/config.yaml"
+      raise
+    end
+
+    ENV["TF_VAR_encryption_keys"] = %Q|[ #{encryption_keys.keys.join(", ")} ]|
```

Comment:
I get this error:

```
[terragrunt] [/project/live/dev/secret-mgmt] 2018/10/18 13:21:31 Running command: terraform apply -auto-approve -input=false -auto-approve                                                           

Error: Cannot parse value for variable encryption_keys ("[default, gcp-stackdriver-export]") as valid HCL: At 1:40: unexpected token while parsing list: IDENT  
```
I'm trying to set the env variable properly

Reply:
Again, the secrets keep reversed at my laptop, but now it fails because the `unless` condition. I added this for debugging:
```
+    secrets_config = YAML.load(File.read("./modules/gcp-secret-mgmt/config.yaml"))
+    secrets_config["encryption_keys"].each do |encryption_key|
+      encryption_keys[encryption_key] = %Q|"#{encryption_key}"|
+    end
+    puts "=="
+    puts collected_secrets.keys
+    puts "=="
+    puts encryption_keys.keys
+    puts "=="
+    unless collected_secrets.keys == encryption_keys.keys
+      puts "ERROR: Secret keys: #{(collected_secrets.keys - encryption_keys.keys).join(", ")} not present in"
+      puts "ERROR: /modules/gcp-secret-mgmt/config.yaml"
+      raise
+    end
```

and I got:

```
==
gcp-stackdriver-export
default
==
default
gcp-stackdriver-export
==
ERROR: Secret keys:  not present in
ERROR: /modules/gcp-secret-mgmt/config.yaml
rake aborted!
```

Reply:
@amatas Yeah, sorry, last commit should fix this issue.